File: solo-leveling-ai-service/src/main/java/com/sleepkqq/sololeveling/ai/kafka/SaveTasksProducer.java
package com.sleepkqq.sololeveling.ai.kafka;

import static com.sleepkqq.sololeveling.avro.constants.KafkaTaskTopics.SAVE_TASKS_TOPIC;

import com.sleepkqq.sololeveling.avro.task.SaveTasksEvent;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Slf4j
@Service
@RequiredArgsConstructor
public class SaveTasksProducer {

  private final KafkaTemplate<String, SaveTasksEvent> kafkaTemplate;

  public void send(SaveTasksEvent event) {
    kafkaTemplate.send(SAVE_TASKS_TOPIC, event);
    log.info("<< Save tasks event sent | transactionId={}", event.getTransactionId());
  }
}

File: solo-leveling-ai-service/src/main/kotlin/com/sleepkqq/sololeveling/ai/kafka/SaveTasksProducer.kt
package com.sleepkqq.sololeveling.ai.kafka

import com.sleepkqq.sololeveling.avro.constants.KafkaTaskTopics
import com.sleepkqq.sololeveling.avro.task.SaveTasksEvent
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class SaveTasksProducer(
	private val kafkaTemplate: KafkaTemplate<String, Any>
) {

	private val log = LoggerFactory.getLogger(javaClass)

	fun send(event: SaveTasksEvent) {
		kafkaTemplate.send(KafkaTaskTopics.SAVE_TASKS_TOPIC, event)
		log.info("<< Save tasks event sent | txId={}", event.txId)
	}
}
